Login/Signup page first commit

Sarnet2020 opened the pull request.

Sarnet2020 committed:
Login/Signup page first commit

elliottallmann commented on gui/app/(tabs)/login.tsx:
```diff
@@ -0,0 +1,108 @@
+// Login Page
+const Login = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
```
We can issue a remember token, caching the plain text password in the state probably isn't a great thing to do.

Sarnet2020 replied:
How can i get the token from the login API?

elliottallmann replied:
Will be covered in #14 , no need to worry about it here.

elliottallmann commented on gui/app/(tabs)/login.tsx:
```diff
@@ -0,0 +1,108 @@
+      <TextInput
+        placeholder="Email"
+        value={email}
+        onChangeText={setEmail}
+        style={styles.input}
+      />
+      <TextInput
+        placeholder="Password"
+        value={password}
+        onChangeText={setPassword}
+        secureTextEntry
+        style={styles.input}
+      />
+      <TextInput
+        placeholder="Confirm Password"
+        value={confirmPassword}
+        onChangeText={setConfirmPassword}
+        secureTextEntry
+        style={styles.input}
+      />
+      <Button title="Sign Up" onPress={handleSignUp} />
+    </View>
+  );
+};
+
+const App = () => {
+  const [isLogin, setIsLogin] = useState(true);
+
+  return (
+    <View style={styles.container}>
+      {isLogin ? <Login /> : <SignUp />}
+      <Text
+        style={styles.toggleText}
+        onPress={() => setIsLogin(!isLogin)}
+      >
+        {isLogin ? 'Need to create an account?' : 'Already have an account?'}
+      </Text>
+    </View>
+  );
+};
```
#18 handles the Sign Up here, I don't think we need to replicate.

The App component should be defined in one central place, that determines if a user has a valid logged-in session or not. It should be in something like /gui/index.tsx, whatever it is that gets served as the default route. That component is what should handle any decision making, and should simply load this component.

Maybe @collwill can provide some insight here, we can talk about it more during our meeting tomorrow.

elliottallmann replied:
I did some more reading on this. I think what the best thing to do is remove the App and SignUp components from here, and let those be handled by what's already done and by #18. 

 IMO, the "Login" form should be a component that lives under something like /gui/components/auth/loginForm.tsx. It would contain any styling appropriate to the actual form, and encapsulate any API calls and validation logic. 

The /gui/app/(tabs)/login.tsx file would position the LoginForm components on the entire page. This way we can pass any props we need to back and forth with the component, but we would be able to keep the display / positioning logic separate from the component functionality logic.

Something like: 
```javascript
import React from 'react';
import {StyleSheet, View, Text } from 'react-native';
import LoginForm from 'app/components/loginForm';

const Login = () => { 

return (
    <View style={styles.container}>
        <LoginForm />
    </View>
);

}
```

And the LoginForm would look something like: 
```javascript

import React from 'react';
import {StyleSheet, View, Text, TextInput, Button } from 'react-native';

const LoginForm = () => {

let isValidForm = false;

const handleLoginRequest = (formData) => {
  if (this.isValidForm) {
    axios.post('/auth/login', formData);
  }
};

let password;

const handlePasswordChange = (password) => {
  this.password = password;
}


return (
    <View style={styles.container}>
        <TextInput 
            placeholder="email"
            value={email}
            onChangeText={setEmail}
            style={styles.input}
        />
       <TextInput
        placeholder="Password"
        value={password}
        onChangeText={handlePasswordChange}
        secureTextEntry
        style={styles.input}
      />
    <Button title="Login" onPress={handleLoginRequest} />
    </View>
  );

};

export default LoginForm;

```

Note that this code probably won't run, but it should give you a good starting point.

Sarnet2020 committed:
Added the validation on login
removed the signup
moved login component file from app/tabs to component/auth

NOTE: waiting for login API for token part.
